src/app/service.ts

```diff
@@ -24,25 +25,53 @@ export const makeApp = ({
 
     await httpHandler.run()
 
-    const messages = await messagesProcessor.load()
-    const verifiedMessages = await messagesProcessor.verify(messages)
+    /**
+     * Appends new messages to messageStorage,
+     * Not modifying the messagesStorage reference
+     *
+     * Notes:
+     *   - new messages are identified by difference of signature
+     */
+    const reloadAndVerifyMessages = async (messagesStorage: MessageStorage) => {
+      logger.info(`Will reload messages`)
+      logger.info(`Existing messages count ${messagesStorage.length}`)
+      const messages = await messagesProcessor.load()
+      const verifiedMessages = await messagesProcessor.verify(messages)
+
+      logger.info(`Existing messages count ${messagesStorage.length}`)
+
+      const appendedMessagesCount =
+        messagesStorage.addMessages(verifiedMessages)
+
+      logger.info(`Appended ${appendedMessagesCount} new messages`)
+    }
+
+    const messageStorage = new MessageStorage()
+    await reloadAndVerifyMessages(messageStorage)
 
     logger.info(
       `Starting, searching only for requests for operator ${OPERATOR_ID}`
     )
 
     logger.info(`Loading initial events for ${BLOCKS_PRELOAD} last blocks`)
-    await job.once({ eventsNumber: BLOCKS_PRELOAD, messages: verifiedMessages })
+    await job.once({
+      eventsNumber: BLOCKS_PRELOAD,
+      messageStorage: messageStorage,
+    })
 
     logger.info(
       `Starting ${
         JOB_INTERVAL / 1000
       } seconds polling for ${BLOCKS_LOOP} last blocks`
     )
 
+    setInterval(() => {
```

Comment:
I would suggest using a similar logic as the above-described `job`. You can connect metrics and logging there.

```ts
const job = makeJobRunner('validator-ejector', {
    config,
    logger,
    metric: metrics.jobDuration,
    handler: jobProcessor.handleJob,
  })
```